apps/openchallenges/infra/cdk/README.md

```diff
@@ -2,51 +2,268 @@
 ```
 
 ### Deploy Infrastructure
 
 Deploy the CDK stacks to AWS:
 
 ```bash
-nx deploy openchallenges-infra-cdk
+# Development
+nx deploy openchallenges-infra-cdk:dev
+
+# Staging
+nx deploy openchallenges-infra-cdk:stage
+
+# Production
+nx deploy openchallenges-infra-cdk:prod
+```
+
+### Destroy Infrastructure
+
+Remove all deployed resources:
+
+```bash
+# Development
+nx destroy openchallenges-infra-cdk:dev
+
+# Staging
+nx destroy openchallenges-infra-cdk:stage
+
+# Production
+nx destroy openchallenges-infra-cdk:prod
+```
+
+## Developer Workflow
+
+### Setting Up Your Development Stack
+
+1. **Set your developer name** in `.env.dev`:
+
+   ```bash
+   echo "DEVELOPER_NAME=your-github-username" >> .env.dev
+   ```
+
+2. **Configure AWS credentials** for the development account:
+
+   ```bash
+   aws configure --profile openchallenges-dev
+   ```
+
+3. **Bootstrap CDK** (first-time only):
+
+   ```bash
+   ENV=dev cdk bootstrap --profile openchallenges-dev
+   ```
+
+4. **Synthesize and deploy**:
+   ```bash
+   nx synth openchallenges-infra-cdk:dev
+   nx deploy openchallenges-infra-cdk:dev
+   ```
+
+### Making Changes
+
+1. Make changes to constructs or stacks in the `shared/` directory
+2. Test synthesis: `nx synth openchallenges-infra-cdk:dev`
+3. Review diff: `nx diff openchallenges-infra-cdk:dev`
+4. Deploy changes: `nx deploy openchallenges-infra-cdk:dev`
+
+### Cleaning Up
+
+When you're done testing, destroy your development stack:
+
+```bash
+nx destroy openchallenges-infra-cdk:dev
 ```
 
-## Environment Configuration
+## Phase 0: Minimal Infrastructure
+
+This initial phase deploys a single S3 bucket to validate the multi-environment deployment workflow.
+
+### Resources Deployed
+
+- **Image Bucket**: S3 bucket for storing images (used by Thumbor service)
+  - Encryption: S3-managed (AES256)
+  - Public access: Blocked
+  - Versioning: Disabled
+  - Naming:
+    - Dev: `openchallenges-dev-{developer-name}-img`
+    - Stage/Prod: `openchallenges-{env}-img`
+
+### CloudFormation Outputs
+
+The bucket stack exports the following values:
+
+- `ImageBucketName`: Name of the S3 bucket
+- `ImageBucketArn`: ARN of the S3 bucket
+
+These outputs can be referenced by other stacks or services.
+
+## Testing
+
+### Run All Tests
+
+```bash
+nx test openchallenges-infra-cdk
+```
+
+### Run Unit Tests Only
+
+```bash
+nx test openchallenges-infra-cdk --testPathPattern=unit
+```
+
+### Run Integration Tests Only
+
+```bash
+nx test openchallenges-infra-cdk --testPathPattern=integration
+```
+
+## Environment Variable Reference
+
+### Required for All Environments
+
+- `ENV`: Environment name (`dev`, `stage`, or `prod`)
+
+### Required for Development
+
+- `DEVELOPER_NAME`: Your identifier for resource naming (alphanumeric, hyphens, underscores only)
+
+### Optional
+
+- `AWS_REGION`: AWS region for deployment (defaults to CDK's default region)
+- `AWS_ACCOUNT`: AWS account ID (defaults to CDK's default account)
```

Comment:
`AWS_PROFILE` might be a better option because i believe it encapsulates both the account and region.